### ISSUE TITLE
dummy procedure argument (fm317 fm328 fm509)

### ISSUE THREAD

**Issue by schweitzpgi** [flang-compiler/f18-llvm-project#286]
function signature generation

**Comment by sscalpone** [comment]
Happens in LAPACK, e.g. `lapack/TESTING/EIG/cdrgsx.f`

```
% cat s1.f90
SUBROUTINE serred (path, nunit)
 LOGICAL sslect
 EXTERNAL :: sslect
 EXTERNAL :: sgees
 CALL sgees(sslect)
 RETURN
END SUBROUTINE
```
```
bbc s1.f90 |& head
/local/home/sjs/mirror/fir-dev-mirror/flang/lib/Lower/ConvertExpr.cpp:356: not yet implemented
UNREACHABLE executed at /local/home/sjs/mirror/fir-dev-mirror/flang/lib/Lower/ConvertExpr.cpp:356!
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
Stack dump:
0.      Program arguments: bbc s1.f90
 #0 0x000055fb577cf9ca llvm::sys::PrintStackTrace(llvm::raw_ostream&) (/local/home/sjs/mirror/install/bin/bbc+0x111c9ca)
 #1 0x000055fb577cd8c4 llvm::sys::RunSignalHandlers() (/local/home/sjs/mirror/install/bin/bbc+0x111a8c4)
 #2 0x000055fb577cda08 SignalHandler(int) (/local/home/sjs/mirror/install/bin/bbc+0x111aa08)
 #3 0x00007f68a574a890 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x12890)
 #4 0x00007f68a4410e97 raise /build/glibc-OTsEL5/glibc-2.27/signal/../sysdeps/unix/sysv/linux/raise.c:51:0
```

**Comment by jeanPerier** [comment]
Dummy procedure argument with procedures not defined in program unit implemented in #314.
fm317 fm328 fm509 now compile.
